proposals/lip-0049.md

```diff
@@ -430,36 +461,65 @@ registrationCCMParamsSchema = {
 * `name`: name registered on the mainchain.
 * `messageFeeTokenID`: ID of the token used for cross-chain message fees.
 
-##### Creating Registration Message
+##### Creation
 
 A registration message is created by the Interoperability module when registering a chain, as specified in [LIP 0043][lip-0043].
 
-##### Executing Registration Message
+##### Verification
+```python
+def verify(ccm: CCM) -> None:
+    if channel(ccm.sendingChainID).inbox.size != 0:
+        raise Exception("Registration message must be the first message in the inbox.")
+    
+    if ccm.status != CCM_STATUS_OK:
+        raise Exception("Registration message must have status OK.")
+    
+    if ownChainAccount.ID != ccm.receivingChainID:
+        raise Exception("Registration message must be sent to the chain account ID of the chain.")
+    
+    if ownChainAccount.name != ccm.params.name:
+        raise Exception("Registration message must contain the name of the registered chain.")
+    
+    if channel(ccm.sendingChainID).messageFeeTokenID != ccm.params.messageFeeTokenID:
+        raise Exception("Registration message must contain the same message fee token ID as the chain account.")
+    
+    networkID = networkID of the chain
+    if ccm.params.networkID != networkID:
+        raise Exception("Registration message must contain the network ID of the registered chain.")
+    
+    # Processing on the mainchain
```

Comment:
The code from here on we would remove with the flat ecosystem. For the current protocol, I wonder what we gain by these checks and whether we could not remove them already.

Reply:
Indeed, but the goal of the previous round of updates was exactly to prepare the interop module to the flat ecosystem. Here I really think we do not lose anything by elaving these checks already here

Reply:
Sounds good.